src/py/flwr/server/superlink/state/utils_test.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utils tests."""
+
+import unittest
+
+from .utils import (
+    convert_sint64_to_uint64,
+    convert_uint64_to_sint64,
+    generate_rand_int_from_bytes,
+)
+
+
+class UtilsTest(unittest.TestCase):
+    """Test utils code."""
+
+    def test_convert_uint64_to_sint64(self) -> None:
+        """Test conversion from uint64 to sint64."""
+        # Test values below 2^63
+        self.assertEqual(convert_uint64_to_sint64(0), 0)
+        self.assertEqual(convert_uint64_to_sint64(2**62), 2**62)
+        self.assertEqual(convert_uint64_to_sint64(2**63 - 1), 2**63 - 1)
+
+        # Test values at and above 2^63
+        self.assertEqual(convert_uint64_to_sint64(2**63), -(2**63))
+        self.assertEqual(convert_uint64_to_sint64(2**63 + 1), -(2**63) + 1)
+        self.assertEqual(convert_uint64_to_sint64(2**64 - 1), -1)
```

Comment:
Perhaps it'd be useful/cleaner to use `parameterized` here (we've to add `parameterized` in our `dev` dependencies), e.g.
```python
    @parameterized.expand(
        [
            # Test values below 2^63
            (0, 0),
            (2**62, 2**62),
            (2**63 - 1, 2**63 - 1),
            # Test values at and above 2^63
            (2**63, -(2**63)),
            (2**63 + 1, -(2**63) + 1),
            (2**64 - 1, -1),
        ]
    )
    def test_convert_uint64_to_sint64(self, before: int, after: int) -> None:
        """Test conversion from uint64 to sint64."""
        self.assertEqual(convert_uint64_to_sint64(before), after)
```
This follows the test structure in Flower Datasets, like [here](https://github.com/adap/flower/blob/630d78775ea5c17e63ff6a486ae5ac6241a66d31/datasets/flwr_datasets/partitioner/dirichlet_partitioner_test.py#L56-L62).